File: frontend/src/app/models/register-request.ts
// Define an interface for user registration requests.
export interface RegisterRequest {

  // Optional field for the user's first name.
  firstname?: string;

  // Optional field for the user's last name.
  lastname?: string;

  // Optional field for the user's email address.
  email?: string;

  // Optional field for the user's chosen password.
  password?: string;

  // Optional field for the user's role (if applicable).
  role?: string;

  // Optional field indicating whether Two-Factor Authentication (2FA) is enabled for the user.
  mfaEnabled?: string;
}

File: backend/src/main/java/com/deliktas/security/auth/RegisterRequest.java
package com.deliktas.security.auth;

import com.deliktas.security.user.Role;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * Represents a request for user registration.
 */
@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class RegisterRequest {

  private String firstname;
  private String lastname;
  private String email;
  private String password;
  private Role role;
  private boolean mfaEnabled;
}
